design.md

```diff
@@ -69,23 +88,27 @@ The client must periodically call a keepalive function in the kernel. If the ker
 
 - There is a file runner interface which specifies:
   - A method to run a script in a repo that accepts a list of objects and returns an object.
+    - The kernel must internally have a method to run a script with a list of devs to implement the gRPC API call to run scripts.
   - A method to load a robot from a robot config file specified as a file specifier and a connection method.
     - A connection method can be a device connection method (e.g. UDP or HID) or a simulator.
-- A script is given a string containing a JSON object. The schema of this object is:
+- When a script is run it is given a string containing a JSON object. The schema of this object is:
   - A dictionary of script environment variables. This is not to be confused with the operating system's environment variables or the JVM's environment variables.
-- Scripts have persistent storage. Scripts can modify files "on disk" and these changes need to make their way to the user's machine. ! RFC: When is this needed? I can't think of a use case.
+- Scripts have persistent storage. There is an interface method in the kernel's API that gives you a `Path` that the script can perform any file io inside. The changes inside this directory will be sync'd to a directory inside the project when the script is finished running. This directory is shared between all scripts.
```

Comment:
Need to spec out the API calls necessary to support this (or is it just more "status message" types in the "run script" streaming call?)

Reply:
Somehow we need to transfer files from client to kernel and back. This feels like something someone else would have already implemented.

Reply:
For MVP, we can create a tarball on the kernel side, send the bytes over, and then unpack the tarball on top of the client's copy of the directory.